src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1047,13 +1023,39 @@ private boolean addEntry(final Map<String, List<MapEntry>> entryMap, final Strin
         return map;
     }
 
-	/**
+    /**
+     *
+     *  validate alias path based on configuration provided
+     */
+
+    protected boolean isValidAliasPath(final String path){
+        if(path == null){
+            throw new IllegalArgumentException("Unexpected null path");
+        }
+
+        // ignore system tree
+        if (path.startsWith(JCR_SYSTEM_PREFIX)){
+            log.debug("loadAliases: Ignoring {}", path);
+            return false;
+        }
+        Set<String> allowedPaths = this.factory.getAllowedAliasPaths();
+        // check allow list
+        if(!allowedPaths.isEmpty()){
```

Comment:
Why the extra check? The next statement will do this for you anyway.

Reply:
if allowed path is empty then all path will be considered and the method should return true
 boolean allowed = allowedPaths.stream().anyMatch(path::startsWith);
This statement will return false if allowedPaths will be empty
That's why an empty check is applied explicitly